src/app/request/request-detail/request-detail.component.html

```diff
@@ -30,4 +30,6 @@ <h1>Request from {{request?.propagator?.id}} </h1>
   </button>
   <button id="deleteBtn" type="button" *ngIf="(isRole('propagator') && currentUserEdit()) || isRole('admin')" [routerLink]="['delete']" class="btn m-1 btn-outline-danger">Delete
   </button>
+  <button id="donateBtn" type="button" *ngIf="isRole('donor') || isRole('admin')" (click)="createDonation(request)" class="btn m-1 btn-outline-success">Donate
```

Comment:
What if the request has been already satisfied by a donation? Should the button be hidden in that case? Or maybe simply completely hide the completed requests from the list?

Reply:
Done, with a test